typescript/web/src/components/import-button/import-images-modal/modal-dropzone/file-statuses.tsx

```diff
@@ -1,68 +1,147 @@
-import { RiImageLine, RiFile3Line } from "react-icons/ri";
+import { FileWithPath } from "react-dropzone";
+import {
+  RiImageLine,
+  RiBracesLine,
+  RiFolderZipLine,
+  RiFile3Line,
+} from "react-icons/ri";
 import { isEmpty } from "lodash/fp";
 import { Box, Text, Flex, useColorModeValue as mode } from "@chakra-ui/react";
 
-import { DroppedFile, UploadStatuses } from "../types";
+import { DroppedFile, UploadInfoRecord, UploadInfo } from "../types";
 import { ImportProgress } from "../import-progress";
 import { ImportError } from "../import-error";
 
+type FileStatusProps = {
+  droppedFile: DroppedFile;
+  index: number;
+  fileUploadInfo?: UploadInfo;
+};
+
+const fileStatusIcon = (file: FileWithPath, hasErrors: boolean) => {
+  if (hasErrors) {
+    return RiFile3Line;
+  }
+  switch (file.type) {
+    case "application/json":
+      return RiBracesLine;
+    case "application/zip":
+      return RiFolderZipLine;
+    default:
+      return RiImageLine;
+  }
+};
+
+const ImportStatus = ({
+  droppedFile,
+  fileUploadInfo,
+}: Omit<FileStatusProps, "index">) => (
+  <Box
+    whiteSpace="nowrap"
+    flex={0}
+    color={mode("gray.400", "gray.600")}
+    fontSize="md"
+    textAlign="right"
+  >
+    {isEmpty(droppedFile.errors) ? (
+      <ImportProgress status={fileUploadInfo?.status || false} />
+    ) : (
+      <ImportError errors={droppedFile.errors} />
+    )}
+  </Box>
+);
+
+const FilePath = ({ droppedFile }: Pick<FileStatusProps, "droppedFile">) => (
+  <Box
+    pr="2"
+    flexGrow={1}
+    flexShrink={1}
+    overflow="hidden"
+    textOverflow="ellipsis"
+    whiteSpace="nowrap"
+  >
+    {droppedFile.file.path}
+  </Box>
+);
+
+const Warnings = ({ warnings }: { warnings: string[] | undefined }) => (
+  <>
```

Comment:
No VStack, ul or table here?

Reply:
there is a `Flex` component in `FileStatus` in which those will go

Reply:
OK fine 👍 Thanks for the answer!